lib/cocoapods/command/repo/add_cdn.rb

```diff
@@ -0,0 +1,55 @@
+module Pod
+  class Command
+    class Repo < Command
+      class AddCDN < Repo
+        self.summary = 'Add a spec repo backed by a CDN'
+
+        self.description = <<-DESC
+          Add `URL` to the local spec-repos directory at `#{Config.instance.repos_dir}`. The
+          remote can later be referred to by `NAME`.
+        DESC
+
+        self.arguments = [
+          CLAide::Argument.new('NAME',   true),
+          CLAide::Argument.new('URL',    true),
+        ]
+
+        def initialize(argv)
+          @name = argv.shift_argument
+          @url = argv.shift_argument
+          super
+        end
+
+        def validate!
+          super
+          unless @name && @url
```

Comment:
both of these checks should already by handled by the `super` call?

Reply:
not sure if they are but `add.rb` command has
```ruby
          if @name == 'master' || @url =~ %r{github.com[:/]+cocoapods/specs}i
            raise Informative,
                  'To setup the master specs repo, please run `pod setup`.'
          end
```

Reply:
i.e it also checks the `@url`

Reply:
I wanted to replicate the logic in the `add` command, but without erroring out on CP/Specs.